R/write_dataset_json.R

```diff
@@ -36,13 +43,20 @@ write_dataset_json <- function(x, file, pretty=TRUE) {
     pretty = pretty)
 
   # Run the validator
-  jsonvalidate::json_validate(js, schema_1_0_0, engine="ajv", error=TRUE)
+  valid <- jsonvalidate::json_validate(js, schema_1_0_0, engine="ajv")
+
+  if (!valid) {
+    stop(paste(
+      "Dataset JSON file is invalid per the JSON schema.",
+      "Use datasetjson::validate_dataset_json() to see details"),
+      call.=FALSE)
+  }
```

Comment:
A user would need all the steps above line 48 to recreate what they need to pass to `datasetjson::validate_dataset_json()` right?  

Should we just go ahead and pass `js` to `datasetjson::validate_dataset_json() ` and return this?

Or maybe leave it as you have it, but ask the user if you want to see the results of `datasetjson::validate_dataset_json() ` and then return if yes.

Reply:
The output of `validate_dataset_json()` is too verbose because it's a big dataframe. So I wanted to make this more intentional instead of printing a whole lot of stuff to the console.